### PR TITLE
Update documentation on Verto 'relative-link' behaviour (fixes #504)

### LINKED ISSUE #504: Update documentation on Verto 'relative-link' behaviour
I've configured the `relative-link` Verto tag to use the following HTML template:

```html
{% autoescape false %}<a href="{{ '{% url' }} {{ link_path }} {{ '%}' }}">{{ text }}</a>{% endautoescape %}
```

So with the following Markdown:

```markdown
You need to find the factors of the number (see [challenge 4.1]('topics:programming_challenge' 'unplugged-programming' 'check-if-divisor')).
```

The following HTML is created:

```html
You need to find the factors of the number (see <a href="/en-gb/topics/unplugged-programming/programming/check-if-divisor">challenge 4.1</a>).
```

This is really useful as the URLs are less likely to break, however we need to explain the basics of the Django URL command to authors.

### PR DESCRIPTION
**Pull request by JackMorganNZ**
Also fixes an internal link.

Fixes #504.

### CODE REVIEW

File: docs/source/author/writing_guide.rst [hunk 1]
```diff
@@ -227,47 +227,62 @@ destination of the link.
...
+  values, and the `Django URL template tag <https://docs.djangoproject.com/en/1.11/ref/templates/builtins/#url>`__.
+  If you are unsure on how to create a link of this type, please contact us
+  and we will be happy to help.
 
-You can refer to a page by writing the page name with ``.html`` at the end.
-The name of a file is defined by it's slug in the configuration files, but
-it helps to have knowledge of the resulting URL path for a file.
-See the examples below:
+Links to pages within the website use the Django ``url`` template tag.
+In order to create an link, use the values that would be given to the template tag.
```
Comment (hayleyavw) [inline docs/source/author/writing_guide.rst]: `create an link` -> `create a link`

**Comment by hayleyavw** [comment]
Slightly confused, the relative-link template doesn't seem to include params?

File: docs/source/author/writing_guide.rst [hunk 2]
```diff
@@ -227,47 +227,62 @@ destination of the link.
...
+This link uses the same syntax as internal links, with appended query parameters.
+Each parameter should be a resource option with its desired value.
+Values given as query parameters override the default values for each resource.
+
+This following Markdown:
 
 .. code-block:: none
 
-  Please [contact us](about/index.html#contact).
+  [Little Red Riding Hood cards]('resources:resource' 'sorting-network-cards'?type=riding_hood)
```
Comment (hayleyavw) [inline docs/source/author/writing_guide.rst]: pls clarify,  the template for relative-link doesn't inlcude params, is `type=riding_hood` just included as part of the path? in which case, how is this parsed (there are quotes around `sorting-network-cards`, what does this mean). Basically, what is this magic?

**Comment by JackMorganNZ** [comment]
Addressed your comments, it was hard to expand on how the links worked without getting overly technical in the authors document (hence the warning message). 

The extra parameters are included in the `{{ link_query }}` in the template:

```html
<a href="{{ '{% url' }} {{ link_path }} {{ '%}' }}{{ link_query }}">{{ text }}</a>
```

File: docs/source/author/writing_guide.rst [hunk 3]
```diff
@@ -266,7 +266,13 @@ Link to resource page with pre-selected options
   If you are unsure on how to create a link of this type, please contact us
   and we will be happy to help.
 
-This link uses the same syntax as internal links, with appended query parameters.
+This link uses the same syntax as internal links, with a query string appended
+at the end.
+The Markdown processor Verto will read the link string and seperate everything
+before and after the ``?`` symbol, and will create a link using the Django URL tag
+with everything before symbol (the same way a normal internal link is created).
```
Comment (hayleyavw) [inline docs/source/author/writing_guide.rst]: `before symbol` -> `before the symbol`

**Pull request merged**